llvm/lib/Target/AMDGPU/SIMachineFunctionInfo.h

```diff
@@ -684,8 +684,8 @@ class SIMachineFunctionInfo final : public AMDGPUMachineFunction,
 
   void setFlag(Register Reg, uint8_t Flag) {
     assert(Reg.isVirtual());
-    if (VRegFlags.inBounds(Reg))
-      VRegFlags[Reg] |= Flag;
+    VRegFlags.grow(Reg);
```

Comment:
I guess, this change is unnecessary. Keep the inbounds check back.
The following change in the other patch would make sure to grow VRegFlags for each virtual register it encountered.
https://github.com/llvm/llvm-project/pull/110228/files#diff-c72079b2a595aca3300d5e3c15d227f81937f2745f7c5494fcf1fe9ba37d8828R1789

Reply:
The MIR function is parsed after parsing the options, so the `noteNewVirtualRegister` callback doesn't take effect.

Reply:
You changed this though?

Reply:
It doesn't take effect here.
Might be better to move it to a separate commit.

Reply:
This change doesn't make sense to me. What will happen to the regular flow when it reaches from MRI createVirtualRegister? Isn't duplicating the size?
https://github.com/llvm/llvm-project/blob/main/llvm/lib/CodeGen/MachineRegisterInfo.cpp#L166

Reply:
I think Register info is an avenue for noting new virtual registers so I've put it here https://github.com/llvm/llvm-project/pull/111634